mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3089 @@
+        FWHM_eV_fit = params[1]
+        amplitude_fit = params[2]
+        prob_parameter_fit = params[3]
+        total_counts_fit = amplitude_fit
+
+        perr = m_binned.np_errors()
+        B_field_fit_err = perr[0]
+        FWHM_eV_fit_err = perr[1]
+        amplitude_fit_err = perr[2]
+        prob_parameter_fit_err = perr[3]
+        total_counts_fit_err = amplitude_fit_err
+    
+        fit_Hz = self.spectrum_func_1(bins_Hz, *params)
+        fit_keV = ComplexLineShapeUtilities.flip_array(fit_Hz)
+        bins_keV = ConversionFunctions.Energy(bins_Hz, B_field_fit)/1000
+        bins_keV = ComplexLineShapeUtilities.flip_array(bins_keV)
+
+        nonzero_bins_index = np.where(data_hist_freq != 0)[0]
+        zero_bins_index = np.where(data_hist_freq == 0)[0]
+        fit_Hz_nonzero = fit_Hz[nonzero_bins_index]  
+        data_Hz_nonzero = data_hist_freq[nonzero_bins_index] 
+        fit_Hz_zero = fit_Hz[zero_bins_index]
+        data_Hz_zero = data_hist_freq[zero_bins_index]
+        reduced_chi2 = self.reduced_chi2_Poisson(data_hist_freq, fit_Hz, number_of_parameters = 4)
+        elapsed = time.time() - t
+        output_string = '\n'
+        output_string += 'Reduced chi^2 = {:.2e}\n'.format(reduced_chi2)
+        output_string += '-----------------\n'
+        output_string += 'B field = {:.8e}'.format(B_field_fit)+' +/- '+ '{:.4e} T\n'.format(B_field_fit_err)
+        output_string += '-----------------\n'
+        output_string += 'Gaussian FWHM = '+str(round(FWHM_eV_fit,2))+' +/- '+str(round(FWHM_eV_fit_err,2))+' eV\n'
+        output_string += '-----------------\n'
+        output_string += 'Amplitude = {}'.format(round(amplitude_fit,2))+' +/- {}'.format(round(amplitude_fit_err,2)) + '\n'
+        output_string += '-----------------\n'
+        output_string += 'Probability parameter \n= ' + "{:.2e}".format(prob_parameter_fit)\
+        +' +/- ' + "{:.2e}".format(prob_parameter_fit_err)+'\n'
+        output_string += '-----------------\n'
+        output_string += 'Fit completed in '+str(round(elapsed,2))+'s'+'\n'
+        dictionary_of_fit_results = {
+        'output_string': output_string,
+        'bins_keV': bins_keV,
+        'fit_keV': fit_keV,
+        'bins_Hz': bins_Hz,
+        'fit_Hz': fit_Hz,
+        'B_field_fit': B_field_fit,
+        'B_field_fit_err': B_field_fit_err,
+        'FWHM_eV_fit': FWHM_eV_fit,
+        'FWHM_eV_fit_err': FWHM_eV_fit_err,
+        'survival_prob_fit': prob_parameter_fit,
+        'survival_prob_fit_err': prob_parameter_fit_err,
+        'amplitude_fit': amplitude_fit,
+        'amplitude_fit_err': amplitude_fit_err,
+        'data_hist_freq': data_hist_freq,
+        'reduced_chi2': reduced_chi2
+        }
+        return dictionary_of_fit_results
+
+    # using simulated resolution, with multi gas scattering, reconstruction eff, without detection eff, has been used in fake data generator. However, without using detection eff is the right option for tritium fake data generation.
```

Comment:
split comments over multiple lines for readability. in general, function descriptions should come after def function_name(). and they should go in between ''' '''.